eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/Imp/EpochSpec.hs

```diff
@@ -18,37 +17,36 @@ import Cardano.Ledger.Shelley.LedgerState (
   utxosGovStateL,
  )
 import Cardano.Ledger.Val (Val (..))
-import Test.Cardano.Ledger.Common (Spec, describe, shouldBe)
+import Test.Cardano.Ledger.Imp.Common
 import Test.Cardano.Ledger.Shelley.ImpTest (
+  ImpTestState,
   ShelleyEraImp,
   getsNES,
-  impIO,
-  itM,
   passEpoch,
   submitTx,
  )
 
 spec ::
   forall era.
   ShelleyEraImp era =>
-  Spec
+  SpecWith (ImpTestState era)
 spec = describe "EPOCH" $ do
-  itM @era "Runs basic transaction" $ do
+  it "Runs basic transaction" $ do
     do
       certState <- getsNES $ nesEsL . esLStateL . lsCertStateL
       govState <- getsNES $ nesEsL . esLStateL . lsUTxOStateL . utxosGovStateL
-      impIO $ totalObligation certState govState `shouldBe` zero
+      totalObligation certState govState `shouldBe` zero
     do
       deposited <- getsNES $ nesEsL . esLStateL . lsUTxOStateL . utxosDepositedL
-      impIO $ deposited `shouldBe` zero
+      deposited `shouldBe` zero
     _ <- submitTx "simple transaction" $ mkBasicTx mkBasicTxBody
     passEpoch
 
-  itM @era "Crosses the epoch boundary" $ do
+  it "Crosses the epoch boundary" $ do
     do
       epoch <- getsNES nesELL
-      impIO $ epoch `shouldBe` 0
```

Comment:
Maybe we could also generalize `shouldReturn` to work with any monad, not just `IO`

Reply:
It is already done, right here: https://github.com/input-output-hk/cardano-ledger/pull/3842/files#diff-2e7f2de716c0d8bb516324b34e09894dda34f4c8c9e1c3aee2078800e729b63dR156

I'll switch those to `shouldReturn`

Reply:
Done